app/Console/Commands/AngularGenerators/CreateSubPageCommand.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Console\Commands\AngularGenerators;
+
+class CreateSubPageCommand extends BaseGeneratorCommand
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'ng:subpage {name} {pageName} {--path=}';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create component files for Angular 2 Sub Page.';
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        $this->validateInput();
+
+           /**
+     * Name of Subpage Component will be {pageName.name}  .. to prevent name collide.
+     */ 
+
+        $componentName = $this->argument('pageName').$this->argument('name');
+
+          /**
+     * Target of Subpage Component will be pages/{pageName}/subpages/{pageName.name} 
+     */ 
+        $targetDir = $this->getTargetDir('pages',$this->argument('pageName').'/subpages', $componentName);

```

Comment:
Sorry sir. I am new here. I wanted to pull it in my master. It came here. I think directory up to sub page will be fine.